src/classes/USER_UserService.cls

```diff
@@ -0,0 +1,271 @@
+    * @description Determines if a User is the owner of the Scheduled Jobs.
+    * @param sysAdminUserIds The User Ids that were modified.
+    * @return void
+    **********************************************************************************************************/
+	public static void handleScheduledJobsSubmittedByInActiveIsFrozenSysAdmin(Set<Id> sysAdminUserIds) {
+		Boolean breakOuterCircle = false;
+
+		List<CronTrigger> scheduledJobs = retrieveScheduledJobs(sysAdminUserIds);
+
+		for(Id currentSysAdminId : sysAdminUserIds) {
+			for(CronTrigger eachScheduledJob : scheduledJobs) {
+				if(currentSysAdminId == eachScheduledJob.OwnerId) {
+					notifySysAdminsAboutInActiveIsFrozenSysAdminUser();
+					breakOuterCircle = true;
+					break;
+				}
+			}
+			if(breakOuterCircle) {
+				break;	
+			}
+		}
+	}
+
+
+	/*********************************************************************************************************
+    * @description Logs an Error__c record and emails the org's System Administrators.
+    * @return void
+    **********************************************************************************************************/
+	@testVisible
+	private static void notifySysAdminsAboutInActiveIsFrozenSysAdminUser() {
+		logInActiveIsFrozenUserError();
+		sendInActiveIsFrozenUserEmail();
+	}
+
+
+	/*********************************************************************************************************
+    * @description Logs an Error__c record.
+    * @return void
+    **********************************************************************************************************/
+	@testVisible
+	private static void logInActiveIsFrozenUserError() {
+		Error__c inActiveIsFrozenError = new Error__c(Error_Type__c = ERROR_TYPE_SCHEDULED_APEX,
+														Object_Type__c = OBJECT_TYPE_USER,
+														Datetime__c = System.now(), 
+														Full_Message__c = ERROR_MESSAGE_SCHEDULED_APEX);
+
+		Database.insert(inActiveIsFrozenError, true);
+	}
+
+
+	/*********************************************************************************************************
+    * @description Emails the org's System Administrators.
+    * @return void
+    **********************************************************************************************************/
+	@testVisible
+	private static void sendInActiveIsFrozenUserEmail() {
+		ERR_Notifier.sendErrorNotifications(CONTEXT_TYPE_USER);
+	}
+
+
+	/*********************************************************************************************************
+    * @description Utility method to retrieve the System Adminstrator's ProfileId.
+    * @return Id The Id of the System Administrator Profile.
+    **********************************************************************************************************/
+	private static Id retrieveSysAdminProfileId() {
+		Id sysAdminProfileId = null; 
+
+		List<Profile> sysAdminProfile = [SELECT Id FROM Profile WHERE Name = :SYSTEM_ADMINISTRATOR];
```

Comment:
what do we do about users who use a cloned profile off of System Adminstrator?  is there any way for us to catch those as well?

Reply:
Presently the code looks for the System Administrator Profile when handling the Error Notification Recipient setting. We could add the functionality to handle cloned profiles. If we do, I'd recommend making the change throughout the code.

Reply:
is there even a foolproof way to know  it was cloned from Sys Admin?